fv3core/stencils/saturation_adjustment.py

```diff
@@ -113,49 +96,41 @@ def qs_table_fn(i):
 @gtscript.function
 def qs_table2_fn(i):
     tem0 = tem_lower(i)
-    table2 = 0.0
     if i < 1600:
         # compute es over ice between - 160 deg c and 0 deg c.
         table2 = table_ice_oneline(tem0)
     else:
         # compute es over water between 0 deg c and 102 deg c.
         table2 = table_vapor_oneline(tem0)
-    table2_m1 = 0.0
-    table2_p1 = 0.0
     # TODO is there way to express the code below with something closer to this?:
     # if i == 1599:
     #    table2 = 0.25 * (table2(i-1) + 2.0 * qs_table_fn(i) + qs_table2_fn(i+1))
     # if i == 1600:
     #    table2 = 0.25 * (table2(i-1) + 2.0 * qs_table_fn(i) + qs_table2_fn(i+1))
-    table = 0.0
+    # table = 0.0
     if i == 1599:
         # table(i)
         table = table_ice_oneline(tem0)
-        tem0 = 253.16 + DELT * (i - 1400)  # tem_upper(i - 1400)
-        # table_vapor_oneline(tem0)
-        table = (0.05 * (TICE - tem0)) * table + (0.05 * (tem0 - 253.16)) * (
-            E00
-            * exp(
-                (DC_VAP * log(tem0 / TICE) + (tem0 - TICE) / (tem0 * TICE) * LV0)
-                / constants.RVGAS
-            )
-        )
+        tem0 = tem_upper(i - 1400)
+        table = (0.05 * (TICE - tem0)) * table + (
+            0.05 * (tem0 - 253.16)
+        ) * table_vapor_oneline(tem0)
         # table2(i - 1)
-        tem0 = TMIN + DELT * 1598  # tem_lower(1598)
+        tem0 = tem_lower(1598)
         table2_m1 = table_ice_oneline(tem0)
         # table2(i + 1)
-        tem0 = TMIN + DELT * 1600  # tem_lower(1600)
+        tem0 = tem_lower(1600)
         table2_p1 = table_vapor_oneline(tem0)
```

Comment:
Can these be merged into 1 statement currently? Or do we need a temporary assigned to pass to the other function?

Reply:
The temporary is still needed since GT4Py doesn't support function calls as arguments to function calls.

` gt4py.frontend.gtscript_frontend.GTScriptSyntaxError: Function calls are not supported in arguments to function calls`

Reply:
Wow, what a helpful error message! this is great.